LNK2001 with optional abort and RID win-x86.

Hello, it seems the optional abort override from 52edb718f49778406ce58c4b5fb6d426c0fb9ef4
leads to a LNK2001, when building for the win-x86 platform.
`platform.obj : error LNK2001: unresolved external symbol _dnne_abort [C:\source\dotnet\DNNE\Lib\Lib.csproj]`

I get this error when building a .NET5 classlib with one export
referencing a DNNE package built from 52edb718f49778406ce58c4b5fb6d426c0fb9ef4 as follows
`C:\source\dotnet\DNNE\Lib> dotnet build -f net5.0   -r win-x86`

Many thanks.

--- a/src/platform/platform.c
+++ b/src/platform/platform.c
@@ -283,12 +283,19 @@ DNNE_API void DNNE_CALLTYPE set_failure_callback(failure_fn cb)
 
 // Provide mechanism for users to override default behavior of certain functions.
 //  - See https://stackoverflow.com/questions/51656838/attribute-weak-and-static-libraries
+//  - See https://devblogs.microsoft.com/oldnewthing/20200731-00/?p=104024
 //  - Use the MSC macro to enable Windows builds without MSVC.
 #ifdef _MSC_VER
     #define DNNE_DEFAULT_IMPL(methodName, ...) default_ ## methodName(__VA_ARGS__)
 
     // List of overridable APIs
-    #pragma comment(linker, "/alternatename:dnne_abort=default_dnne_abort")
+    // Note the default calling convention is cdecl on x86 for DNNE_APIs.
+    // This means, on x86, we mangle the alternative name in the linker command.
+    #ifdef _M_IX86
+        #pragma comment(linker, "/alternatename:_dnne_abort=_default_dnne_abort")
+    #else
+        #pragma comment(linker, "/alternatename:dnne_abort=default_dnne_abort")
+    #endif
 #else
     #define DNNE_DEFAULT_IMPL(methodName, ...) __attribute__((weak)) methodName(__VA_ARGS__)
 #endif

